nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/repository/FileSystemRepository.java

```diff
@@ -149,6 +154,10 @@ public FileSystemRepository(final NiFiProperties nifiProperties) throws IOExcept
         for (final Path path : fileRespositoryPaths.values()) {
             Files.createDirectories(path);
         }
+        this.maxFlowFilesPerClaim = nifiProperties.getMaxFlowFilesPerClaim();
+        this.writableClaimQueue  = new LinkedBlockingQueue<>(maxFlowFilesPerClaim);
+        final String maxAppendableClaimSize = nifiProperties.getMaxAppendableClaimSize();
+        this.maxAppendableClaimLength = DataUnit.parseDataSize(maxAppendableClaimSize, DataUnit.B).intValue();
```

Comment:
In my opinion, I don't see anything wrong with a static cap, as long as it's reasonably big enough (the 100 MB choice).